lib/active_merchant/billing/gateways/checkout_v2.rb

```diff
@@ -128,19 +132,23 @@ def add_transaction_data(post, options = {})
       end
 
       def add_3ds(post, options)
-        if options[:three_d_secure]
+        if options[:three_d_secure] || options[:execute_threed]
```

Comment:
Suggestion: For clarity remove the nested conditional putting the condition that applies to both options[:three_d_secure] and options[:execute_threed] at the top of the method. Placing the specific configuration of  options[:three_d_secure] below it. Something like -

```rb
def add_3ds(post, options)
  if options[:three_d_secure] || options[:execute_threed]
    post[:'3ds'] = {}
    post[:'3ds'][:enabled] = true
    post[:success_url] = options[:callback_url] if options[:callback_url]
    post[:failure_url] = options[:callback_url] if options[:callback_url]
  end

  if options[:three_d_secure]
    post[:'3ds'][:eci] = options[:three_d_secure][:eci] if options[:three_d_secure][:eci]
    post[:'3ds'][:cryptogram] = options[:three_d_secure][:cavv] if options[:three_d_secure][:cavv]
    post[:'3ds'][:version] = options[:three_d_secure][:version] if options[:three_d_secure][:version]
    post[:'3ds'][:xid] = options[:three_d_secure][:ds_transaction_id] || options[:three_d_secure][:xid]
  end
end
```

Thoughts?

Reply:
Definitely makes things more explicit